src/smallcommands.ts

```diff
@@ -39,7 +41,51 @@ function applyTextEdit(we) {
 //     }
 // }
 
+async function newJuliaFile(uri?: vscode.Uri) {
+    if (uri) {
+        const stat = await vscode.workspace.fs.stat(uri)
+        const dir = stat.type === vscode.FileType.Directory ? uri.fsPath : path.dirname(uri.fsPath)
+        const givenPath = await vscode.window.showInputBox({
+            value: dir,
+            valueSelection: [dir.length, dir.length], // no selection
+            prompt: 'Enter a file path to be created',
+            validateInput: async input => {
+                const givenPath = vscode.Uri.file(input).fsPath
+                const exist = await fs.exists(givenPath)
+                if (exist) { return `${givenPath} already exists` }
+                const givenDir = path.dirname(givenPath)
+                const dirExist = await fs.exists(givenDir)
+                if (!dirExist) { return `Directory ${givenDir} doesn't exist` }
+                return undefined // valid
+            }
+        })
+        if (!givenPath) { return } // canceled, etc
+        const targetUri = vscode.Uri.file(givenPath)
+        try {
+            await fs.writeTextFile(targetUri.fsPath, '')
+            vscode.workspace.openTextDocument(targetUri)
+                .then(document => {
+                    return vscode.languages.setTextDocumentLanguage(document, 'julia')
+                })
+                .then(document => {
+                    vscode.window.showTextDocument(document)
+                })
+        } catch (err) {
+            vscode.window.showErrorMessage(`Failed to create ${targetUri.fsPath}`)
+        }
```

Comment:
Not sure this part will be really useful (`File: New File` command with file extension to be `.jl` is more intuitive way to create an julia file imho). But this way allows us to create a file with whatever file extension while setting its language to be julia.
It would be great if vscode offers some API to interact with the Explorer UI, but I couldn't find. that.